flocker/ca/functional/test_validation.py

```diff
@@ -318,3 +330,62 @@ class RESTAPIContextFactoryValidationTests(
     """
     Tests for the context factory that validates REST API clients.
     """
+
+
+class RequestsTests(TestCase):
+    """
+    Tests for the ``requests`` library's ability to talk to REST API TLS
+    server.
+    """
+    def assert_requests_validates(self, hostname, cert_name):
+        """
+        Assert that ``requests`` succeeds in doing a TLS query to our server.
+
+        :param hostname: Hostname to connect to.
+        :param cert_name: Which attribute of the ``ControlSet`` to use.
+
+        :return: ``Deferred`` firing when test is done.
+        """
+        _, port = find_free_port()
+        ca_set, _ = get_credential_sets()
+        context_factory = rest_api_context_factory(
+            ca_certificate=ca_set.root.credential.certificate,
+            control_credential=getattr(ca_set, cert_name))
+        start_tls_server(self, port, context_factory)
+
+        certs = FilePath(self.mktemp())
+        certs.makedirs()
+        ca_set.copy_to(certs, user=True)
+
+        def req():
+            return requests.get(
+                "https://{}:{}/".format(hostname, port),
+                headers={"connection": "close"},
+                stream=False,
+                cert=(certs.child(b"user.crt").path,
+                      certs.child(b"user.key").path),
+                verify=certs.child(b"cluster.crt").path).content
+        pool = ThreadPool()
+        pool.start()
+        self.addCleanup(pool.stop)
+        d = deferToThreadPool(reactor, pool, req)
+        d.addCallback(self.assertEqual, EXPECTED_STRING)
+        return d
```

Comment:
We can't really do much to help users validate their custom 3rd party clients except suggest logging (and provide good docs and maybe client libraries).

I will improve logging on server, at least (we can do that in the context factory verify callback I think).